bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatAccountHandler.java

```diff
@@ -0,0 +1,449 @@
+import org.openhab.binding.millheat.internal.dto.AbstractResponse;
+import org.openhab.binding.millheat.internal.dto.DeviceDTO;
+import org.openhab.binding.millheat.internal.dto.GetHomesRequest;
+import org.openhab.binding.millheat.internal.dto.GetHomesResponse;
+import org.openhab.binding.millheat.internal.dto.GetIndependentDevicesByHomeRequest;
+import org.openhab.binding.millheat.internal.dto.GetIndependentDevicesByHomeResponse;
+import org.openhab.binding.millheat.internal.dto.HomeDTO;
+import org.openhab.binding.millheat.internal.dto.LoginRequest;
+import org.openhab.binding.millheat.internal.dto.LoginResponse;
+import org.openhab.binding.millheat.internal.dto.RoomDTO;
+import org.openhab.binding.millheat.internal.dto.SelectDeviceByRoomRequest;
+import org.openhab.binding.millheat.internal.dto.SelectDeviceByRoomResponse;
+import org.openhab.binding.millheat.internal.dto.SelectRoomByHomeRequest;
+import org.openhab.binding.millheat.internal.dto.SelectRoomByHomeResponse;
+import org.openhab.binding.millheat.internal.dto.SetDeviceTempRequest;
+import org.openhab.binding.millheat.internal.dto.SetRoomTempRequest;
+import org.openhab.binding.millheat.internal.dto.SetRoomTempResponse;
+import org.openhab.binding.millheat.internal.model.Heater;
+import org.openhab.binding.millheat.internal.model.Home;
+import org.openhab.binding.millheat.internal.model.MillheatModel;
+import org.openhab.binding.millheat.internal.model.ModeType;
+import org.openhab.binding.millheat.internal.model.Room;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link MillheatAccountHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arne Seime - Initial contribution
+ */
+public class MillheatAccountHandler extends BaseBridgeHandler {
+    private static final int MIN_TIME_BETWEEEN_MODEL_UPDATES_MS = 30_000;
+    private static final int NUM_NONCE_CHARS = 16;
+    private static final String CONTENT_TYPE = "application/x-zc-object";
+    public static String API_ENDPOINT_1 = "https://eurouter.ablecloud.cn:9005/zc-account/v1/";
+    public static String API_ENDPOINT_2 = "https://eurouter.ablecloud.cn:9005/millService/v1/";
+    private static final String ALLOWED_NONCE_CHARACTERS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private static final String REQUEST_TIMEOUT = "300";
+    private final Logger logger = LoggerFactory.getLogger(MillheatAccountHandler.class);
+    private String userId;
+    private String token;
+    private final HttpClient httpClient;
+    private RequestLogger requestLogger;
+    private MillheatDiscoveryService discoveryService;
+    private Gson gson;
+    private MillheatModel model = new MillheatModel(0);
+    private @Nullable ScheduledFuture<?> statusFuture;
```

Comment:
```suggestion
    private @NonNullByDefault({}) ScheduledFuture<?> statusFuture;
```
Than you don't need to suppress the null warnings. Using of `@NonNullByDefault` like this can be done on fields that are considered not to be null by how the architecture works. In this case the `initialize` is always called by the framework before something else.

Reply:
Done